src/main/java/com/aws/greengrass/secretmanager/SecretManager.java

```diff
@@ -333,26 +343,52 @@ public String validateSecretId(String secretId) throws GetSecretException {
         if (Utils.isEmpty(secretId)) {
             throw new GetSecretException(400, "SecretId absent in the request");
         }
-        try {
-            return getArnFromCache(secretId);
-        } catch (GetSecretException e) {
+        String arn = getArnFromCache(secretId);
+        if (!isSecretPresentInCache(arn)) {
             try {
                 reloadCache();
-                return getArnFromCache(secretId); // throw GetSecretException if the secret doesn't exist
-            } catch (SecretManagerException ex) {
-                throw new GetSecretException(404, secretNotFoundErr + secretId);
+            } catch (SecretManagerException e) {
+                // TODO: Improve return code where device is offline and SM is unable to load cache from disk
+                logger.atWarn().setCause(e).log("Unable to load secrets from cache");
             }
         }
+        return arn;
     }
 
-    private String getArnFromCache(String secretId) throws GetSecretException {
+    private boolean isSecretIdArn(final String secretId) {
+        return Pattern.matches(VALID_SECRET_ARN_PATTERN, secretId);
+    }
+
+    private boolean doesSecretNameMatchWithConfiguredArn(String secretName, String secretArn) {
+        String regex = "^arn:aws:secretsmanager:[a-z0-9-]+:\\d{12}:secret:"
+                + "(?i)" + Pattern.quote(secretName)
+                + "-[A-Za-z0-9]{6}(?::[A-Za-z0-9/_+=.@-]+)?$";
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

These configs are flagged under GS-Tech Static config campaign and is considered to be a violation. Please migrate these configs